### DISCUSSION TITLE
Sudden error

### DISCUSSION THREAD

**Discussion by C8opmBM** [auanasgheps/snapraid-aio-script#134] [Q&A] [answered]
Hello and thanks for your amazing kit which I've been using for years!!!

So today I woke up to an email error, I tried to see if there's any changes or errors, I checked the file and logs in OMV, disks are ok, I wanted to post here in case you have any idea before taking it to the omv forums.
Somehow there's a missing parity message. I checked that drive and opened the config, but all seems ok
`Missing 'parity' specification in '/etc/snapraid/omv-snapraid-114a88d2-53ed-11ed-8eee-b3f2573b9c38.conf'`

Any ideas?
PS: when I run the command alias, there's an initial error thrown, see below : `grep: /etc/snapraid.conf: No such file or directory` 
Could this be related?

Thanks!!

![see](https://github.com/user-attachments/assets/09ebf983-349e-457b-acba-6e1559044379)

**Comment by auanasgheps** [comment]
Hi there!

Since a while, the OMV Plugin for SnapRAID has changed few things: the config files are now dynamic. There might be or not a file in /etc/snapraid.conf.

The script was updated to dynamically pick the new configuration files, and you can see it's doing it:
`/etc/snapraid/omv-snapraid-114a88d2-53ed-11ed-8eee-b3f2573b9c38.conf`

Can you paste the content of the file here?

The error you're getting seems to be related to SnapRAID itself. 

I suspect that the migration from the classic /etc/snapraid.conf to the newly generated file didn't work out well

**Comment by C8opmBM** [comment]
Yes, I did check the conf, didn't see anything out of the ordinary.
Also posted in omv forums, and detailed there the configuration.
https://forum.openmediavault.org/index.php?thread/54636-snapraid-sudden-error/


This is the content of `/etc/snapraid/omv-snapraid-114a88d2-53ed-11ed-8eee-b3f2573b9c38.conf`


```
# This file is auto-generated by openmediavault (https://www.openmediavault.org)
# WARNING: Do not edit this file, your changes will get lost.

autosave 0



# drives
#####################################################################
# OMV-Name: data1  Drive Label: 
content /srv/dev-disk-by-uuid-e2541e00-15ba-4d4a-9f3f-e87e5fb7439e/snapraid.content
data data1 /srv/dev-disk-by-uuid-e2541e00-15ba-4d4a-9f3f-e87e5fb7439e

#####################################################################
# OMV-Name: data2  Drive Label: Media2
content /srv/dev-disk-by-uuid-eba46a1c-e2b3-4b25-bd53-c0e23e077568/snapraid.content
data data2 /srv/dev-disk-by-uuid-eba46a1c-e2b3-4b25-bd53-c0e23e077568

#####################################################################
# OMV-Name: data3  Drive Label: 
content /srv/dev-disk-by-uuid-ce0a0393-9b5b-47ac-a3ae-422a8e983f8f/snapraid.content
data data3 /srv/dev-disk-by-uuid-ce0a0393-9b5b-47ac-a3ae-422a8e983f8f

#####################################################################
# OMV-Name: parity1  Drive Label: 
3-parity /srv/dev-disk-by-uuid-49ec29ad-d80d-42cc-aa36-96f03c4261be/snapraid.3-parity



exclude *.unrecoverable
exclude lost+found/
exclude aquota.user
exclude aquota.group
exclude /tmp/
exclude .content
exclude *.bak
exclude /snapraid.conf*
exclude /usenet/
exclude /torrents/
exclude *.unrecoverable
exclude lost+found/
exclude aquota.user
exclude aquota.group
exclude /tmp/
exclude .content
exclude *.bak
exclude /snapraid.conf*
exclude /.snapshots/
```

**Comment by C8opmBM** [comment]
I think I may have spotted the error, but I dont know how to persistently modify the config.

The output in mail is:
`ERROR - Parity file (/srv/dev-disk-by-uuid-49ec29ad-d80d-42cc-aa36-96f03c4261be/snapraid.3-parity) not found!`
however there's indeed no `snapraid.3-parity` but a `snapraid.parity` as it should be. I don't know from where the 3 comes from.

So how can I edit the /`etc/snapraid/omv-snapraid-114a88d2-53ed-11ed-8eee-b3f2573b9c38.conf`

from 

```
#####################################################################
# OMV-Name: parity1  Drive Label: 
3-parity /srv/dev-disk-by-uuid-49ec29ad-d80d-42cc-aa36-96f03c4261be/snapraid.3-parity
```
to 

```
#####################################################################
# OMV-Name: parity1  Drive Label: 
parity /srv/dev-disk-by-uuid-49ec29ad-d80d-42cc-aa36-96f03c4261be/snapraid.parity
```

Could this be the fix?

**Reply by auanasgheps** [reply]
You can update this by re-configuring the SnapRAID parity/data drives in the OMV GUI at the SnapRAID section.
Any manual change to this file will be overridden.

**Comment by C8opmBM** [comment, marked as answer]
Thanks, solved it, somehow the 3 there was the error. I dirty upgraded long ago from omv 6 and now snapraid was updated and borked my old config.

Thanks man!
